Mathlib/Algebra/Ring/Equiv.lean

```diff
@@ -839,6 +840,11 @@ protected theorem map_pow (f : R ≃+* S) (a) : ∀ n : ℕ, f (a ^ n) = f a ^ n
   map_pow f a
 #align ring_equiv.map_pow RingEquiv.map_pow
 
+protected theorem isUnit_iff (f : R ≃+* S) {a} : IsUnit a ↔ IsUnit (f a) := by
+  refine ⟨fun a_1 => IsUnit.map f a_1, ?_⟩
+  convert IsUnit.map f.symm
+  simp only [RingEquiv.symm_apply_apply]
```

Comment:
Well, we can for `MonoidHomClass` because `IsUnit` is not defined for `Mul` classes.

I did it above, and am not very happy about putting this theorem here. I tried to put it inside `Group\Equiv\Basic.lean`, where I think it should belong, but this creates import loops.